packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -1589,70 +1588,70 @@ class TransformLayer extends OffsetLayer {
   ///
   /// The [transform] property must be non-null before the compositing phase of
   /// the pipeline.
-  Matrix4 get transform => _transform;
-  Matrix4 _transform;
-  set transform(Matrix4 value) {
+  Matrix4? get transform => _transform;
+  Matrix4? _transform;
+  set transform(Matrix4? value) {
     assert(value != null);
-    assert(value.storage.every((double component) => component.isFinite));
+    assert(value!.storage.every((double component) => component.isFinite));
     if (value == _transform)
       return;
     _transform = value;
     _inverseDirty = true;
     markNeedsAddToScene();
   }
 
-  Matrix4 _lastEffectiveTransform;
-  Matrix4 _invertedTransform;
+  Matrix4? _lastEffectiveTransform;
+  Matrix4? _invertedTransform;
   bool _inverseDirty = true;
 
   @override
   void addToScene(ui.SceneBuilder builder, [ Offset layerOffset = Offset.zero ]) {
     assert(transform != null);
-    _lastEffectiveTransform = transform;
+    _lastEffectiveTransform = transform!;
     final Offset totalOffset = offset + layerOffset;
     if (totalOffset != Offset.zero) {
       _lastEffectiveTransform = Matrix4.translationValues(totalOffset.dx, totalOffset.dy, 0.0)
-        ..multiply(_lastEffectiveTransform);
+        ..multiply(_lastEffectiveTransform!);
     }
     engineLayer = builder.pushTransform(
-      _lastEffectiveTransform.storage,
+      _lastEffectiveTransform!.storage,
       oldLayer: _engineLayer as ui.TransformEngineLayer,
```

Comment:
shouldn't that be `_engineLayer as ui.TransformEngineLayer?`?

Reply:
Done